src/main/java/de/tum/in/www1/artemis/service/connectors/RemoteArtemisServiceConnector.java

```diff
@@ -0,0 +1,70 @@
+package de.tum.in.www1.artemis.service.connectors;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.exception.NetworkingError;
+
+class RemoteArtemisServiceConnector<RequestType, ResponseType> {
```

Comment:
I think it is very nice that you took your time to make this reusable.

To enforce the reuse it would be good to add documentation to this class and write a couple of test cases.

Reply:
✅